### PR TITLE
Setup and cleanup host dependencies during the build

### PR DESCRIPTION
**Pull request by giordano**
Companion PR for BinaryBuilderBase: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/43.  CC: @vchuravy 

NOTE: needs tests specifically for `HostBuildDependency`!

### CODE REVIEW

File: Manifest.toml [hunk 1]
```diff
@@ -28,9 +28,9 @@ uuid = "2a0f44e3-6c83-55bd-87e4-b1978d98bd5f"
 
 [[BinaryBuilderBase]]
 deps = ["CodecZlib", "Downloads", "InteractiveUtils", "JSON", "LibGit2", "Libdl", "Logging", "OutputCollectors", "Pkg", "Random", "SHA", "SimpleBufferStream", "TOML", "Tar", "UUIDs", "p7zip_jll"]
-git-tree-sha1 = "b89a4d163105a65e4e3ad0f211e4e68d7dead95f"
-repo-rev = "master"
-repo-url = "https://github.com/JuliaPackaging/BinaryBuilderBase.jl.git"
+git-tree-sha1 = "2c3a44de64fa99991299b32af05995013621c7e7"
+repo-rev = "mg/hostdeps"
+repo-url = "https://github.com/giordano/BinaryBuilderBase.jl"
```
Comment (giordano) [inline Manifest.toml:33]: Revert these before merging!!

**Comment by giordano** [comment]
There are still 3 places in `src/wizard/interactive_build.jl` where I need to add the platform argument to `cleanup_dependencies`.  It's a bit tricky because we don't have the concrete platforms at those point, in some cases they're complicated to get

**Comment by giordano** [comment]
Existing tests are passing :tada: failure in documentation is expected until https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/43 is merged

**Review by staticfloat — APPROVED** [review]

File: src/AutoBuild.jl [hunk 2]
```diff
@@ -659,10 +659,13 @@ function autobuild(dir::AbstractString,
 
         prefix = setup_workspace(
             build_path,
-            source_files;
+            source_files,
+            concrete_platform,
+            default_host_platform;
```
[the remark this answers is not in the document]
Comment (giordano) [reply src/AutoBuild.jl:664]: You must have missed https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/2365e03352b76bba8054ebabd87d4c829b758437/src/Runner.jl#L757-L761 :smile:

**Pull request merged**